lib/Sabberworm/CSS/RuleSet/DeclarationBlock.php

```diff
@@ -591,15 +593,40 @@ public function createFontShorthand() {
 	}
 
 	public function __toString() {
-		return $this->render(new \Sabberworm\CSS\OutputFormat());
+		return $this->render(new OutputFormat());
 	}
 
-	public function render(\Sabberworm\CSS\OutputFormat $oOutputFormat) {
+	/**
+	 * {@inheritdoc}
+	 */
+	public function render(OutputFormat $oOutputFormat) {
 		if(count($this->aSelectors) === 0) {
 			// If all the selectors have been removed, this declaration block becomes invalid
 			throw new OutputException("Attempt to print declaration block with missing selector", $this->iLineNo);
 		}
-		$sResult = $oOutputFormat->implode($oOutputFormat->spaceBeforeSelectorSeparator() . ',' . $oOutputFormat->spaceAfterSelectorSeparator(), $this->aSelectors) . $oOutputFormat->spaceBeforeOpeningBrace() . '{';
+
+		$sResult = '';
+
+		// render comments
+		if ($oOutputFormat->getKeepComments()) {
+			$comments = $this->getCommentsBefore();
+			if (!empty($comments)) {
+				$sResult .= implode(
+					'',
+					array_map(
+						function (Comment $comment) use ($oOutputFormat) {
+							return '/*' . $comment->getComment() . "*/\n";
```

Comment:
There is already a `render` method on `Comment` that does this. I think you can change that to respect the `KeepComments` option and then you could replace this whole block with

```php
$sResult .= $oOutputFormat->implode('', $this->getCommentsBefore());
```